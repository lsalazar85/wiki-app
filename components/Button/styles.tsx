import styled from 'styled-components';
import colors from '../../styles/colors';

export const ButtonElement = styled.div`
  width: 150px;
  background: ${colors.Beige};
  color: ${colors.Black};
  padding: 0.5rem 0;
  border-radius: 1rem;
  text-align: center;

  &:hover {
    background: ${colors.Salmon};
    transition: 0.5s;
  }

  a {
    display: block;
    width: 100%;
    height: 100%;
    color: ${colors.Black};
    text-decoration: none;
  }
`;

export const ButtonForm = styled.button`
  width: 150px;
  background: ${colors.Beige};
  color: ${colors.Black};
  padding: 0.7rem 0;
  font-size: 0.9rem;
  border-radius: 1rem;
  text-align: center;

  &:hover {
    background: ${colors.Salmon};
    transition: 0.5s;
  }
`;
