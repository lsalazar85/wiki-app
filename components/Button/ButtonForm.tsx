
import { ButtonForm } from './styles';
import { ButtonFormProps } from '../../interfaces';

const Button = ({ text }: ButtonFormProps ) => (
  <ButtonForm>{text}</ButtonForm>
);

export default Button;
