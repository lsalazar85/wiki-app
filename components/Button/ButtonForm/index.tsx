
import { ButtonFormWrapper } from '../styles';
import { ButtonFormProps } from '../../../interfaces';

const ButtonForm = ({ text }: ButtonFormProps ) => (
  <ButtonFormWrapper>{text}</ButtonFormWrapper>
);

export default ButtonForm;
