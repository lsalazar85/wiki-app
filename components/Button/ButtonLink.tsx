import Link from 'next/link';
import { ButtonElement } from './styles';
import { ButtonProps } from '../../interfaces';

const ButtonLink = ({ text, href }: ButtonProps) => (
    <ButtonElement>
        <Link href={href}>{text}</Link>
    </ButtonElement>
);

export default ButtonLink;
