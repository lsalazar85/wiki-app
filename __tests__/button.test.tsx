/**
 * @jest-environment jsdom
 */


import { render  } from '@testing-library/react'
import Button from "../components/Button/ButtonForm";

describe('Button', () => {
    it('should render successfully', () => {
        const { asFragment } = render(<Button text='Button' fn={jest.fn()} />)

        expect(asFragment()).toMatchSnapshot();
        expect(asFragment()).toBeDefined()
    })
})
