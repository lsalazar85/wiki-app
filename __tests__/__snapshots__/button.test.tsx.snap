// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render successfully 1`] = `
<DocumentFragment>
  <button
    class="styles__ButtonContainer-sc-9v32al-0 kRkStR"
  >
    Button
  </button>
</DocumentFragment>
`;
